WordPress/src/main/java/org/wordpress/android/util/ViewUtils.kt

```diff
@@ -13,3 +15,22 @@ fun View.redirectContextClickToLongPressListener() {
         this.setOnContextClickListener { it.performLongClick() }
     }
 }
+
+fun View.expandTouchTargetArea(dps: Int, heightOnly: Boolean = false) {
```

Comment:
This works but I am concerned about maintainability. The dimensions are now stored in constants scattered [throughout the code](https://github.com/wordpress-mobile/WordPress-Android/blob/1d950a0daa4d9b7869f554c21d577f3ff3bc3a4c/WordPress/src/main/java/org/wordpress/android/ui/reader/adapters/ReaderPostAdapter.java#L126-L128) instead of `dimens.xml`. Granted, this only affects the touch area and not the UI itself. Still, I wonder if we should be using paddings/margins instead. 

@malinajirka I would like to hear your thoughts on this.

Reply:
So if we use margins/padding that will actually cause design changes which we don't want at this point. 

> This works but I am concerned about maintainability. The dimensions are now stored in constants scattered throughout the code instead of dimens.xml.

I wholeheartedly agree with this, some possible solutions in my mind are at runtime we could grab the measured height of the view convert it from `pixels` to `dp` and then subtract that from the expected touch target so the whole function would be dynamic. That's a possible approach I didn't explore because so far this is how I have seen most people solve the issue but if we can find an enhanced approach then that's even better. If it helps, we could have a centralized location for the constants themselves and encapsulate that in the function. Let me know. 

For the most part, what I have realized with the `TouchDelegate` is it only applies the expanded area/padding only with the parent of the target view so even if you set a value that is above the amount available by the parent it won't work incorrectly but for clarity and to be concise only the necessary amount was added to the targets.

Reply:
Hmm, an interesting issue, thanks for pinging me ;).

- Could we perhaps just move the constants to `dimens.xml` and pass the resource id to the method - it has access to a context, so it can load it, right? Wdyt?

- I'm also wondering how will this method behave when there are two clickable items right next to each other. Will the one which expands the touchTargetRect later win? I don't have a full context here, but I think that solving this using padding might be safer. Wdyt?

>I wholeheartedly agree with this, some possible solutions in my mind are at runtime we could grab the measured height of the view convert it from pixels to dp and then subtract that from the expected touch target so the whole function would be dynamic.

This doesn't sound bad either, however, I'd probably explore the simple solutions first. This would probably require using something like View.OnLayoutChangeListener and I'm always careful before I decide to use them.


SideNote: We might want to consider using `dpsHeight`, `dpsWidth` as parameters instead of using the boolean flag. What if the view was a rectangle and we needed a different dimensions for height vs width. Wdyt?

Reply:
Hi Jirka! Thanks for your feedback :) 

> Could we perhaps just move the constants to dimens.xml and pass the resource id to the method - it has access to a context, so it can load it, right? Wdyt?

Yes, this could be done for sure. 

> I'm also wondering how will this method behave when there are two clickable items right next to each other. Will the one which expands the touchTargetRect later win? I don't have a full context here, but I think that solving this using padding might be safer. Wdyt?

So far based on my tests,  if two clickable items are beside each other the touchable area only extends to the available space in the parent but not onto the other child view. The behavior of the API really allows existing behavior to remain unimpacted. Again, padding is great 😄 but it would break the UI so this is the safest bet in most cases. 

> This doesn't sound bad either, however, I'd probably explore the simple solutions first. This would probably require using something like View.OnLayoutChangeListener and I'm always careful before I decide to use them.

Yes! That's why I didn't attempt to try it because detect finalized view measurements can be tricky and involve a lot of additional logic. 

> SideNote: We might want to consider using dpsHeight, dpsWidth as parameters instead of using the boolean flag. What if the view was a rectangle and we needed a different dimensions for height vs width. Wdyt?

We could but I don't really think it's necessary. 

1) We are going to be enforcing that most layouts are designed with accessibility in mind with touch targets of 48dp or more by default. 

2) The method would signature get another parameter that might not be used alot because in most cases the touch target changes on views need to be both width and height and then with layouts most are `match_parent` for their width so only the height needs expansion. That's my view based on what I have seen so far. 

Wdyt right back at you 😅

Reply:
Thanks for the reply ;)

>Again, padding is great 😄 but it would break the UI so this is the safest bet in most cases.

I understand it'd change the UI a bit in some cases. I don't think it'd be a problem, but we can use the proposed approach since it seems it doesn't have any big drawbacks.

>>SideNote: We might want to consider using dpsHeight, dpsWidth as parameters instead of using the boolean flag. What if the view was a rectangle and we needed a different dimensions for height vs width. Wdyt?

>We could but I don't really think it's necessary.

👍 


All in all, I'd consider moving the constants to `dimens.xml` and pass either the size or resourceId to the method.

Reply:
> I understand it'd change the UI a bit in some cases. I don't think it'd be a problem, but we can use the proposed approach since it seems it doesn't have any big drawbacks.

Duly noted, I will weigh both approaches and if it's case like below where the UI definitely needs improvement then that can be done but yes, if not then the touch target approach will be utilized.

> All in all, I'd consider moving the constants to dimens.xml and pass either the size or resourceId to the method.

Yes, indeed. Thanks for this suggestion. It greatly improves readability and the footprint of the changes.

Reply:
@malinajirka made the suggest changes 🙏

Reply:
Awesome, thank you! ;) @shiki could you please review the changes - I'm not sure if it's the only thing blocking the merge or if there is something else. It looks good to me from a quick glance :).